xdsl/interpreters/pdl_interp.py

```diff
@@ -403,6 +412,26 @@ def call_func(
 
         return interpreter.run_ssacfg_region(op.body, args, op.sym_name.data)
 
+    @impl_terminator(pdl_interp.ApplyConstraintOp)
+    def run_apply_constraint(
+        self,
+        interpreter: Interpreter,
+        op: pdl_interp.ApplyConstraintOp,
+        args: tuple[Any, ...],
+    ) -> tuple[Any, ...]:
+        assert len(args) == 1
+        constraint_name = op.constraint_name.data
+
+        if constraint_name not in self.native_constraints:
+            raise InterpretationError(f"Unknown constraint function: {constraint_name}")
+        passed, *results = self.native_constraints[constraint_name](
+            *args,
+        )
+        is_negated = op.is_negated.value
+        terminator = op.true_dest if passed != bool(is_negated) else op.false_dest
```

Comment:
```suggestion
        terminator = op.true_dest if passed != is_negated else op.false_dest
```
Does this work?

Reply:
I can drop `.value`, but without the `bool` a false value is represented as `-1`.

Reply:
ah of course